presto-jdbc/src/main/java/com/facebook/presto/jdbc/ColumnInfo.java

```diff
@@ -134,10 +137,28 @@ public static void setTypeInfo(Builder builder, TypeSignature type)
                 builder.setColumnDisplaySize(24);
                 break;
             case "varchar":
-                builder.setSigned(true);
-                builder.setPrecision(VARCHAR_MAX);
+                if ((parameters.size() == 1) && parameters.get(0).isLongLiteral()) {
```

Comment:
We don't need these checks -- the signature is guaranteed for the data type. See how decimal is done.